pkg/test/e2e/default_app_catalog_tests/application_test.go

```diff
@@ -0,0 +1,366 @@
+	"k8c.io/kubermatic/v2/pkg/util/wait"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlruntimelog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	applicationInstallationName string
+	applicationName             string
+	applicationNamespace        string
+	applicationVersion          string
+	key                         string
+	names                       string
+	defaultValuesBlock          string
+	credentials                 jig.AWSCredentials
+	logOptions                  = utils.DefaultLogOptions
+)
+
+const (
+	projectName = "def-app-catalog-test-project"
+)
+
+func init() {
+	flag.StringVar(&applicationInstallationName, "application-installation-name", "", "name of the ApplicationInstallation object")
+	flag.StringVar(&applicationName, "application-name", "", "name of an application from the default app catalog")
+	flag.StringVar(&applicationNamespace, "application-namespace", "", "namespace of an application from the default app catalog")
+	flag.StringVar(&applicationVersion, "application-version", "", "version of an application from the default app catalog")
+	flag.StringVar(&key, "app-label-key", "", "a Kubernetes recommended label used for identifying the name of an application")
+	flag.StringVar(&names, "names", "", "names of the pods of an application from the default app catalog")
+	flag.StringVar(&defaultValuesBlock, "default-values-block", "", "default values block of an application from the default app catalog")
+	credentials.AddFlags(flag.CommandLine)
+	jig.AddFlags(flag.CommandLine)
+	logOptions.AddFlags(flag.CommandLine)
+}
+
+func TestClusters(t *testing.T) {
+	if applicationName == "" || applicationVersion == "" || applicationInstallationName == "" || applicationNamespace == "" || key == "" || names == "" {
+		return
+	}
+
+	rawLog := log.NewFromOptions(logOptions)
+	logger := rawLog.Sugar()
+	ctx := context.Background()
+
+	if err := credentials.Parse(); err != nil {
+		t.Fatalf("Failed to get credentials: %v", err)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		t.Fatalf("failed to build config: %v", err)
+	}
+
+	seedClient, err := ctrlruntimeclient.New(config, ctrlruntimeclient.Options{})
+	if err != nil {
+		t.Fatalf("failed to build ctrlruntime client: %v", err)
+	}
+
+	// set the logger used by sigs.k8s.io/controller-runtime
+	ctrlruntimelog.SetLogger(zapr.NewLogger(rawLog.WithOptions(zap.AddCallerSkip(1))))
+
+	client, cleanup, tLogger, err := createUserCluster(ctx, t, logger, seedClient)
+	if cleanup != nil {
+		defer cleanup()
+	}
+
+	if err != nil {
+		t.Fatalf("failed to create user cluster: %v", err)
+	}
+
+	testUserCluster(ctx, t, tLogger, client)
+}
+
+//gocyclo:ignore
```

Comment:
I think the linter is trying to tell us something. Why are just we just ignoring it?